src/NzbDrone.Core/DecisionEngine/Specifications/ReleaseRestrictionsSpecification.cs

```diff
@@ -30,6 +30,13 @@ public virtual Decision IsSatisfiedBy(RemoteEpisode subject, SearchCriteriaBase
             _logger.Debug("Checking if release meets restrictions: {0}", subject);
 
             var title = subject.Release.Title;
+            var origin = subject.Release.Origin ?? "";
+            var source = subject.Release.Source ?? "";
+            var container = subject.Release.Container ?? "";
+            var codec = subject.Release.Codec ?? "";
+
+            title += "." + origin + "." + source + "." + container + "." + codec;
```

Comment:
If this information isn't part of release title, I'm not sure periods make the most sense since they're unlikely scene releases.

I still don't think appending it to the title is the best course of action, maybe checking each individually is too much, so testing them all together would be better.

Use string interpolation (`var str = $"{origin}.{source}"` etc) for combining these, it looks like that'll also handle null values.

Reply:
Anything we're checking will need to be logged for troubleshooting purposes later, if a release is rejected or accepted for something not in the release name it'll be confusing to troubleshoot.

Reply:
I didn't think about logging. Maybe checking individually would be the best approach and then you can log based on what matched the restriction.